moveos/moveos-stdlib/moveos-stdlib/sources/object.move

```diff
@@ -22,14 +23,10 @@ module moveos_std::object {
         value: T,
     }
 
-    #[private_generics(T)]
     /// Create a new object, the object is owned by `owner`
-    /// The private generic is indicate the T should be defined in the same module as the caller. This is ensured by the verifier.
-    public fun new<T: key>(ctx: &mut TxContext, owner: address, value: T): Object<T> {
+    public(friend) fun new<T: key>(ctx: &mut TxContext, owner: address, value: T): Object<T> {
         let id = tx_context::fresh_object_id(ctx);
         let obj = Object<T>{id, value, owner};
-        //TODO after add event, then remove the debug info
-        debug::print(&obj);
```

Comment:
Consider adding an event to new object ?  Otherwise the object id cannot be obtained in some test cases.

Reply:
We can return the ChangeSet to the client. The client can get all state changes from the ChangeSet.